consensus/hotstuff/pacemaker.go

```diff
@@ -40,29 +40,27 @@ type PaceMaker interface {
 	// CurView returns the current view.
 	CurView() uint64
 
-	// UpdateCurViewWithQC will check if the given QC will allow PaceMaker to fast
-	// forward to QC.view+1. If PaceMaker incremented the current View, a NewViewEvent will be returned.
-	UpdateCurViewWithQC(qc *flow.QuorumCertificate) (*model.NewViewEvent, bool)
+	// ProcessQC will check if the given QC will allow PaceMaker to fast-forward to QC.view+1.
+	// If PaceMaker incremented the current View, a NewViewEvent will be returned.
+	ProcessQC(qc *flow.QuorumCertificate) (*model.NewViewEvent, bool)
 
-	// UpdateCurViewWithBlock will check if the given block will allow PaceMaker to fast forward
-	// to the BlockProposal's view. If yes, the PaceMaker will update it's internal value for
-	// CurView and return a NewViewEvent.
-	//
-	// The parameter `nextPrimary` indicates to the PaceMaker whether or not this replica is the
-	// primary for the NEXT view taking block.view as reference.
-	// True corresponds to this replica being the next primary.
-	UpdateCurViewWithBlock(block *model.Block, isLeaderForNextView bool) (*model.NewViewEvent, bool)
+	// ProcessTC will check if the given TC will allow PaceMaker to fast-forward to TC.view+1.
+	// If PaceMaker incremented the current View, a NewViewEvent will be returned.
+	ProcessTC(tc *flow.TimeoutCertificate) (*model.NewViewEvent, bool)
 
 	// TimeoutChannel returns the timeout channel for the CURRENTLY ACTIVE timeout.
-	// Each time the pace maker starts a new timeout, this channel is replaced.
+	// Each time the pacemaker starts a new timeout, this channel is replaced.
 	TimeoutChannel() <-chan time.Time
 
 	// OnTimeout is called when a timeout, which was previously created by the PaceMaker, has
-	// looped through the event loop. When used correctly, OnTimeout will always return
-	// a NewViewEvent.
+	// looped through the event loop. Triggering a timeout will result with high probability in creating a TimeoutObject.
 	// It is the responsibility of the calling code to ensure that NO STALE timeouts are
 	// delivered to the PaceMaker.
-	OnTimeout() *model.NewViewEvent
+	OnTimeout()
```

Comment:
Should this return a `TimeoutObject`? 

Currently, in the [Implementation Draft](https://www.notion.so/dapperlabs/Implementation-Draft-f654eaa990954508908304ba03428a24#7056470494294bbaa207d94fd6684fcc), `OnTimeout` directly broadcasts the TO. But in HotStuff currently, only `EventHandler` has access to the `Communicator` component for sending messages. We can keep this responsibility split if `Pacemaker` returns the TO for `EventHandler` to broadcast.

Reply:
Originally I wanted `PaceMaker` to broadcast it, but after your comment and thinking a bit more I think it's better to do this in `EventHandler`. Thanks for suggestion.